lark/parse_tree_builder.py

```diff
@@ -21,6 +21,16 @@ def __call__(self, children):
             return self.node_builder(children)
 
 
+
+def _pp_get_meta(children):
+    for c in children:
+        if isinstance(c, Tree):
+            if not c.meta.empty:
+                return c.meta
+        elif isinstance(c, Token):
+            if not c.isspace():     # Don't regard whitespace-only tokens
```

Comment:
It might be interesting to toggle this behavior: Sometimes one wants to not ignore Whit space (extreme example: The language White space) But I agree that ignore WS should be the default behavior. Note that this is backwards incompatible.

Reply:
I do wonder if anyone would actually need to account for the whitespace, when trying to find where nodes start and end.

I can't really imagine the situation, tbh. But perhaps a good compromise would be only to disregard whitespace tokens that being with an underscore?

So `_NEWLINE` won't be counted, but `NEWLINE` will be. I'm not sure how intuitive that is to the casual user. But at the same time, how often will anyone care?

Of course we can add yet another switch to Lark for this, but as you know, I'd like to avoid that as much as possible.

Or a directive like `%ignore_positions WS _NEWLINE`. Which also seems a bit excessive for this case.

Reply:
One usecase where this change is potentially even breaking is when you want to classify what Tree node a specific position in the source file belongs to. If you as example have a dedicated `ws_or_comment: (WS|comment)+` rule then this change might make for confusing boundaries.

We don't need to add a new flag. We can just do what we do in so many other options as well: overload the keyword to something like `False`, `True`, `"keep_ws"`, `"ignore_ws"`.

Does Propagate position act on `%ignore` terminals and `_UNDERSCORE` Terminals? I thought they have already been thrown away by that point. That is at least the behavior I would have expected.

I would certainly not add a new flag/new syntax.

Reply:
> We can just do what we do in so many other options as well: overload the keyword

Hah, good point. Didn't even consider it.

Yeah, that might be the best way. That way "ignore_ws" can be a special behavior, and we won't have to break compatibility.

One question still remains - should it only discard _WHITESPACE, or also WHITESPACE? (underscore or not)

> Does Propagate position act on %ignore terminals and _UNDERSCORE

No, it ignores `%ignore` terminals, but does include _UNDERSCORE terminals, as they might include _COLON or _DEFINE etc.

Reply:
If it normally behaves the same on with no regard to `_`, then it should continue to do so. If someone comes to us and wants propagate_position to ignore whitespaces at this specific point and not at others, we can still add a new option then.